### ISSUE TITLE
[`preserveAllComments`] SWC incorrectly marks __esModule property as pure 

### ISSUE THREAD

**Issue by skutyepov** [swc-project/swc#6849]
### Describe the bug

TL;DR; SWC adds `/*#__PURE__*/ ` above `Object.defineProperty(exports, "__esModule", ({value: true}));`, which leads to removal of this code by minifiers, and therefore causes runtime errors. 

Consider the following input

```
// my-module.js
export default async (meaningOfLife) => { return meaningOfLife ? 42 : -42; };
```

```
// index.js
import mod from './my-module';
// after minification, below will throw X.default is not a function
mod();
```

and SWC config:

```
      {
        test: /\.js$/,
        use: {
          loader: 'swc-loader',
          options: {
            module: {
              type: 'commonjs',
            },
            jsc: {
              preserveAllComments: true
            },
          },
        },
      },
```

Note: `preserveAllComments:true` seems to be the culprit here.

With such setup SWC will incorrectly add `/*#__PURE__*/ ` above the code where `exports.__esModule` is defined. This leads to removal of this code by minifiers later. 
<img width="447" alt="image" src="https://user-images.githubusercontent.com/86685583/214394592-e05d62e6-a891-499e-8b4e-54d0102285bc.png">

Note(2):
This seems to happen only for async arrow functions, which are exported as default. It doesn't happen for sync arrow functions, or `async function() {}`.

Min repro is available here: https://github.com/skutyepov/swc-esmodule-bug-min-repro/tree/main 

### Config

```json
{
  "module": {
    "type": "commonjs"
  },
  "jsc": {
    "preserveAllComments": true
  }
}
```

### Expected behavior

SWC doesn't add `/*#__PURE__*/ ` above exports.__esModule

### Actual behavior

SWC incorrectly adds `/*#__PURE__*/ ` here
<img width="442" alt="image" src="https://user-images.githubusercontent.com/86685583/214396715-05facdc5-fbb4-4bd6-9432-8948f9954ca4.png">

### Version

1.3.28

**Comment by apathetic** [comment]
I recently experienced this as well.  I ended up refactoring a module so as not to export an async arrow fn, but realize that may not be feasible in other contexts

**Comment by skutyepov** [comment]
Can't reproduce with the latest swc (1.11.13). I think this issue can be closed now.

**Comment by kdy1** [comment]
Thanks!
